### PR TITLE
measure_homodyne returns complex (tfbackend)

### PR DESCRIPTION
**Pull request by ziofil**
Fixes the issue where the `sympy.lambdify` function failed to multiply a complex value by a float eager tensor. Solved by having measure_homodyne cast to tf.complex64 (in the tfbackend)

### CODE REVIEW

File: strawberryfields/backends/tfbackend/circuit.py [hunk 1]
```diff
@@ -794,7 +794,7 @@ def measure_homodyne(self, phi, mode, select=None, **kwargs):
 
             self._update_state(new_state)
 
-        return meas_result
+        return tf.cast(meas_result, dtype=tf.complex64)
```
Comment (co9olguy) [inline strawberryfields/backends/tfbackend/circuit.py]:
```suggestion
        return tf.cast(meas_result, dtype=ops.def_type)
```
for compatibility if we change things in future

**Comment by co9olguy** [comment]
Seems to be 2 failing tests :thinking:

**Comment by josh146** [comment]
Note: @smite and I had a conversation about this in #softwaredev, and it looks like this isn't the cause of the issue --- instead, a quickfix is required within the parameter evaluation code

**Comment by josh146** [comment]
Closing this for now, as this is superceded by #320

File: strawberryfields/backends/tfbackend/circuit.py [hunk 2]
```diff
@@ -794,7 +794,7 @@ def measure_homodyne(self, phi, mode, select=None, **kwargs):
 
             self._update_state(new_state)
 
-        return meas_result
+        return tf.cast(meas_result, dtype=ops.def_type)
```
Comment (co9olguy) [inline strawberryfields/backends/tfbackend/circuit.py]: @josh146 @ziofil Please make sure this suggestion makes it to the branch that is taking over

Comment (josh146) [reply strawberryfields/backends/tfbackend/circuit.py]: Done!

**Pull request closed without merging**